### DISCUSSION TITLE
Streaming response body

### DISCUSSION THREAD

**Discussion by extend2to3meters** [tauri-apps/tauri#3138] [Q&A]
I'm trying to stream dynamic data through a response like [this](https://stackoverflow.com/questions/56435409/how-do-i-stream-a-hyper-requests-body-from-a-slow-processing-side-thread-that-p). I have looked at the "streaming" example but supporting range requests is not what I'm looking to achieve.

While embedding a server would work, it might be overkill for this single purpose. I would like to see if this was possible without too much trouble with the existing tauri ResponseBuilder/Response.

I'm not familiar with rust but possibly something like
```rust
let mut response = ResponseBuilder::new();
let stream = STREAM_HERE
response.body(stream)
// http connection kept open and sending chunks until stream has finished
```

**Comment by extend2to3meters** [comment]
Upon closer inspection, it looks like the wry's response is fixed as a Vec\<u8> so I don't think there is a simple solution. I will be embedding a http server instead but I'm still interested in any possible solutions without it.

**Reply by evbo** [reply]
@JonasKruckenberg I saw your posting this morning on streamed requests but don't have permission to respond here:
https://discord.com/channels/616186924390023171/807549941936816148/1011202155014799421

Is your PR related to this? I am performing long-running SQLite queries whereby I convert the streaming row iterator to a fallible iterator and then serialize it as one big JSON object that the `Tauri` `Command` then sends back to my JS frontend, but I would love to immediately return data _pre_ all of that and serialize individual row records as they come for my frontend to display in closer-to-realtime.

There is an example as referenced for streaming video content, but it's not clear how this would extend to asynchronous `Tauri` commands (or maybe `Events`?):
https://github.com/tauri-apps/tauri/blob/dev/examples/streaming/index.html

Would be great to know for my sake and perhaps this discussion how to best support gradually returning data in a streaming fashion.

**Reply by FabianLars** [reply]
(you can get write access for the contributor channels by reacting to the last message in the #rules channel)

**Reply by JonasKruckenberg** [reply]
Yes, the idea is to allow bi-directional streaming of request content and response content in the url scheme handler. 

**But a big disclaimer:**

The changes I've been making to wry to enable this are highly experimental right now and will not be merged for a couple months I guess (we're once again blocked by webkit2gtk not having the necessary APIs) but if you're fine with those restrictions and *really really need* to stream responses, hit me up on discord and I can help you figure out how to integrate the changes.

PS: Do expect breaking changes at any time though while we figure out how to best do this.

PPS: I have only implemented this on macOS so far 😁

**Reply by evbo** [reply]
Thank you both and I'm totally fine waiting. Heck, it's easy enough to fake reactive design with spinners and incremental network requests (multiple smaller queries vs 1 big one, sacrificing some performance) in the meantime, but longer term I'm really interested in what you come up with :)

**Reply by zak-cloudnc** [reply]
Has there been any movement on this restriction? The api seems to be only able to respond with `http::Response<Into<Cow<'static, [u8]>>>` but I may be missing something https://docs.rs/tauri/latest/src/tauri/app.rs.html#2313-2316
